src/systempolicy/systemPolicy.go

```diff
@@ -936,6 +938,17 @@ func PopulateSystemPoliciesFromSystemLogs(sysLogs []types.KnoxSystemLog) []types
 		// get k8s pods
 		pods := cluster.GetPods(clusterName)
 
+		// check for vm type and if exist add default to podname
```

Comment:
Lets keep this loop before/outside of current for loop. Operate directly on `clusteredLogs`.

Reply:
As per my understanding, we needed this loop here. Consider in case of VM, the pods will be nil. If pods are nil, clustering of logs on pod based will not work. Hence this loop will append `default` to the pods  and have the pods non nil.